### PR TITLE
feat: add an ec2 instance

### PR DESCRIPTION
**Pull request by adamhjk**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -93,4 +91,18 @@ resource "aws_route_table_association" "public_subnet_asso" {
...
+
+resource "aws_key_pair" "fedora-pair" {
+    key_name = "fedora-pair"
+    public_key = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDRCT1V+9Ss9P0QwZse4EMhqKvR0SsFuohkp+F20HsFujpuVydvr7+OdWnNgBsk9XVHDvKFLQtfY5hMy8f8NTqnLrZIcMP44vznjs9JvdsIlSvXu++tJn2FgayqnFtqYbjNB8GBamlHT7beBT+LbDeFji4j3RaHaBjScIFgEmkh9Q3FgPpYueRya14t+NFbhD/MxTiY/SSrOXfa5//eP0SaQs9nZjGA3ERX/GthQMp44fV6CznT9H9AVp5heQJKAEbt9pI5j05APeWhmkXHuOeOiYYldZ9uF51woQ7PMotc1CZFZv35DsPVxxZeaqNgS5OmadLkA5fz0h2ehvnksJmuibaPLXD0KLjJr0j63VitNteeTqEnsw6/4TQCXDOy4D7YWetKbpsy0nAj6YdD08KV+InPPd+5HBq6xg5I6MMan0DWi2BgcqfHcqjrbiE9w7SKU3i0Uj72nUVRYDw6huBQFCcK8RIiJSpbdrV6rtK9thyN24su7f2mMAQzX9jaD6rcjcn+nYYWOZGXQmUeO0MqfbzuWyQsrjhlmheiZDrzQaC5PVsgcI6luo3wnv2tf6Hf11JI5dZO6fIHHr16fsmf2iUaNjjJys0cxCIQ6v8mrnONj+ftF1ISt6bgg0y0UyyYuG01JlDyucOV5fvE6+qKg4WQ+nPO6kE5F6NToxGV6w== adam@stalecoffee.org"
+}
+
+resource "aws_instance" "fedora" {
+    ami = "ami-06b7f026f48cd3c6b"
+    instance_type = "t2.micro"
+    key_name = aws_key_pair.fedora-pair.key_name
```
Comment (bobotclown42) [inline main.tf:104]: Without a key_name, you won't be able to log in!

**Review by bobotclown42 — CHANGES_REQUESTED** [review]
You won't be able to log in!

File: main.tf (hunk 1 above)
Comment (adamhjk) [reply main.tf:104]: good catch! I'll fix it.

**Review by bobotclown42 — APPROVED** [review]
LGTM

**Pull request merged**